File: DP/jump_one_leg.py
import sys

rl = lambda : sys.stdin.readline().rstrip()
# tc = int(input())
tc = int(rl())
testcase = []
for i in range(tc):
    # size = int(input())
    size = int(rl())
    game_table = []
    for row in range(size):
        # game_table.append(list(map(int, input().split())))
        game_table.append(list(map(int, rl().split())))
    testcase.append([game_table, size])

for TC in testcase:
    table = TC[0]
    n = TC[1]
    path = [[0] * n for _ in range(n)]

    path[0][0] = 1

    for i in range(n):
        for row in range(i+1):
            for col in range(i+1):
                # 1 인지 확인
                if path[row][col] == 1:

                    # 해당 열에서 값을 사용하여 오른쪽 혹은 아래로 이동
                    val = table[row][col]
                    # 벗어날 경우 아웃
                    # 안 벗어나면 1 체크
                    # 아래
                    if row + val <= i:
                        path[row + val][col] = 1
                    # 오른쪽
                    if col + val <= i:
                        path[row][col + val] = 1
    if path[n - 1][n - 1] == 1:
        print("YES")
    else:
        print("NO")

File: DP/jump_one_leg.cpp
#include <iostream>

using namespace std;

int main(void) {
    int tc;
    cin >> tc;

    int *ans = new int[tc];
    for (int i = 0; i < tc; i++) {
        int size;
        cin >> size;

        int **game_table;
        game_table = new int *[size];
        for (int j = 0; j < size; j++) {
            game_table[j] = new int[size];
        }
        for (int k = 0; k < size; k++) {
            for (int l = 0; l < size; l++) {
                cin >> game_table[k][l];
            }
        }

        int **path;
        path = new int *[size];
        for (int j = 0; j < size; j++) {
            path[j] = new int[size];
        }
        path[0][0] = 1;

        for (int j = 0; j < size; j++) {
            for (int row = 0; row < j + 1; row++) {
                for (int col = 0; col < j + 1; col++) {
                    if (path[row][col] == 1) {
                        int val = game_table[row][col];

                        if (row + val <= j) {
                            path[row + val][col] = 1;
                        }
                        if (col + val <= j) {
                            path[row][col + val] = 1;
                        }
                    }
                }
            }
        }
        if (path[size - 1][size - 1] == 1) {
            ans[i] = 1;
        } else {
            ans[i] = 0;
        }
    }

    for (int i = 0; i < tc; i++) {
        if (ans[i] == 1) {
            cout << "YES" << endl;
        } else {
            cout << "NO" << endl;
        }
    }
}
